R/validate_target_pr.R

```diff
@@ -0,0 +1,366 @@
+      # target_type left in the hub, skip the dataset check. Otherwise will return
+      # a check error
+      if (any(pr_df$oracle_output) && !skip_dataset_check) {
+        oo_dataset_vals <- try_check(
+          validate_target_dataset(
+            hub_path,
+            target_type = "oracle-output",
+            validations_cfg_path = validations_cfg_path,
+            round_id = round_id
+          ),
+          file_path = "oracle-output"
+        )
+      } else {
+        oo_dataset_vals <- NULL
+      }
+      # Validate individual time-series files but only if not removed
+      oo_files <- pr_df$rel_path[
+        pr_df$oracle_output & pr_df$status != "removed"
+      ]
+      if (length(oo_files) > 0L) {
+        oo_vals <- purrr::map(
+          oo_files,
+          ~ validate_target_submission(
+            hub_path,
+            file_path = .x,
+            target_type = "oracle-output",
+            date_col = date_col,
+            na = na,
+            output_type_id_datatype = output_type_id_datatype,
+            validations_cfg_path = validations_cfg_path,
+            round_id = round_id,
+            skip_check_config = TRUE
+          )
+        ) %>%
+          purrr::list_flatten() %>%
+          as_target_validations()
+      } else {
+        oo_vals <- NULL
+      }
+
+      validations <- combine(
+        validations,
+        file_modifications,
+        ts_dataset_vals,
+        ts_vals,
+        oo_dataset_vals,
+        oo_vals
+      )
+    },
+    error = function(e) {
+      # This handler is used when an unrecoverable error is thrown. This can
+      # happen when, e.g., the csv file cannot be parsed by read_csv(). In this
+      # situation, we want to output all the validations until this point plus
+      # this "unrecoverable" error.
+      e <- capture_exec_error(
+        file_path = gh_repo,
+        msg = conditionMessage(e)
+      )
+      validations <<- combine(
+        validations,
+        new_hub_validations(e)
+      )
+    }
+  )
+  return(validations)
+}
+
+
+is_target_type_single_file <- function(
+  path,
+  target_type = c("time-series", "oracle-output")
+) {
+  target_type <- rlang::arg_match(target_type)
+
+  file_name <- fs::path_ext_remove(fs::path_file(path)) # filename without extension
+  ext <- fs::path_ext(path) # extension
+  file_name == target_type && ext %in% c("csv", "parquet")
+}
+
+is_target_type_dir_file <- function(
+  path,
+  target_type = c("time-series", "oracle-output")
+) {
+  target_type <- rlang::arg_match(target_type)
+  fs::path_has_parent(path, fs::path("target-data", target_type)) &&
+    stringr::str_detect(path, "README", negate = TRUE)
+}
+
+is_target_type_file <- function(
+  path,
+  target_type = c("time-series", "oracle-output")
+) {
+  target_type <- rlang::arg_match(target_type)
+  is_target_type_single_file(path, target_type) ||
+    is_target_type_dir_file(path, target_type)
+}
```

Comment:
Not a blocking comment, but I am not sure I understand the logic here. Would it be possible to explain the logic, especially on the `is_target_type_single_file()`